akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala

```diff
@@ -499,32 +500,55 @@ class Http2ServerSpec extends AkkaSpec with WithInPendingUntilFixed with Eventua
         error should ===(ErrorCode.FRAME_SIZE_ERROR)
       }
 
-      "received SETTINGS_MAX_FRAME_SIZE" in pending
+      "received SETTINGS_MAX_FRAME_SIZE should cause outgoing DATA to be chunked up into at-most-that-size parts " in new TestSetup with RequestResponseProbes {
+        val maxSize = Math.pow(2, 15).toInt // 32768, valid value (between 2^14 and 2^14 - 1)
+        sendSETTING(SettingIdentifier.SETTINGS_MAX_FRAME_SIZE, maxSize)
+
+        expectSettingsAck()
+
+        sendWINDOW_UPDATE(0, maxSize * 10) // make sure we can receive such large response, on connection
+
+        sendHEADERS(1, endStream = false, endHeaders = true, HPackSpecExamples.C41FirstRequestWithHuffman)
+        sendWINDOW_UPDATE(1, maxSize * 5) // make sure we can receive such large response, on this stream
+
+        val theTooLargeByteString = ByteString("x" * (maxSize * 2))
+        val tooLargeEntity = Source.single(theTooLargeByteString)
+        val tooLargeResponse = HttpResponse(entity = HttpEntity(ContentTypes.`application/octet-stream`, tooLargeEntity))
+        emitResponse(1, tooLargeResponse)
+
+        expectHeaderBlock(1, endStream = false)
+        // we receive the DATA in 2 parts, since the ByteString does not fit in a single frame
+        val d1 = expectDATA(1, endStream = false, numBytes = maxSize)
+        val d2 = expectDATA(1, endStream = true, numBytes = maxSize)
+        d1.toList should have length maxSize
+        d2.toList should have length maxSize
+        (d1 ++ d2) should ===(theTooLargeByteString) // makes sure we received the parts in the right order
+      }
 
       "received SETTINGS_MAX_CONCURRENT_STREAMS" in pending
 
       "received SETTINGS_HEADER_TABLE_SIZE" in new TestSetup with RequestResponseProbes {
-        sendSETTING(SettingIdentifier.SETTINGS_HEADER_TABLE_SIZE, Math.pow(2, 15).toInt) // 32768, valid value (between 2^14 and 2^14 - 1)
+        sendSETTING(SettingIdentifier.SETTINGS_HEADER_TABLE_SIZE, Math.pow(2, 15).toInt) // 32768, valid value (between 2^14 and 2^24 - 1)
 
         expectSettingsAck() // TODO check that the setting was indeed applied
       }
     }
 
     "support low-level features" should {
-      "respond to PING frames (spec 6_7)" in new TestSetup with RequestResponseProbes {
```

Comment:
Btw. this was on purpose because the jenkins JUnit report file parser otherwise does stupid things and the test tree doesn't work properly in the UI. Let's keep it with the dot and think about how to fix it otherwise.